File: domain/src/test/java/com/viewnext/domain/usecase/GetDetallesUseCaseTest.java
package com.viewnext.domain.usecase;

import com.viewnext.domain.model.Detalles;
import com.viewnext.domain.repository.DetallesCallback;
import com.viewnext.domain.repository.GetDetallesRepository;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyNoMoreInteractions;

import java.util.List;

public class GetDetallesUseCaseTest {

    @Mock
    private GetDetallesRepository mockRepository;

    private GetDetallesUseCase useCase;

    @Before
    public void setUp() {
        MockitoAnnotations.openMocks(this);
        useCase = new GetDetallesUseCase(mockRepository);
    }

    @Test
    public void refreshDetalles_callsRepositoryRefresh() {
        // Arrange
        // @Before

        // Act
        useCase.refreshDetalles(new DetallesCallback<>() {
            @Override
            public void onSuccess(List<Detalles> detalles) {
            }

            @Override
            public void onFailure(Throwable t) {
            }
        });

        // Assert
        verify(mockRepository).refreshDetalles(any(DetallesCallback.class));
        verifyNoMoreInteractions(mockRepository);
    }
}

File: domain/src/test/java/com/viewnext/domain/usecase/GetDetallesUseCaseTest.kt
package com.viewnext.domain.usecase

import com.viewnext.domain.model.Detalles
import com.viewnext.domain.repository.DetallesCallback
import com.viewnext.domain.repository.GetDetallesRepository
import org.junit.Before
import org.junit.Test
import org.mockito.ArgumentMatchers.any
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.Mockito.verifyNoMoreInteractions
import org.mockito.MockitoAnnotations

class GetDetallesUseCaseTest {
    @Mock
    private lateinit var mockRepository: GetDetallesRepository
    private lateinit var useCase: GetDetallesUseCase

    @Before
    fun setUp() {
        MockitoAnnotations.openMocks(this)
        useCase = GetDetallesUseCase(mockRepository)
    }

    @Test
    fun refreshDetalles_callsRepositoryRefresh() {
        // Arrange
        // @Before

        // Act
        useCase.refreshDetalles(object : DetallesCallback<MutableList<Detalles?>?> {
            override fun onSuccess(result: MutableList<Detalles?>?) {
            }

            override fun onFailure(error: Throwable?) {
            }
        })

        // Assert
        verify(mockRepository).refreshDetalles(any())
        verifyNoMoreInteractions(mockRepository)
    }
}